pkg/liquidity-mining/test/GaugeRelativeWeightCap.test.ts

```diff
@@ -70,21 +69,49 @@ describe('GaugeRelativeWeightCap', () => {
     await vault.grantPermissionsGlobally([action], admin);
   });
 
+  enum GaugeKind {
+    LiquidityGauge,
+    StakelessGauge,
+  }
+
   describe('LiquidityGaugeV5', () => {
-    beforeEach('use liquidity gauge factory', () => {
-      factory = liquidityGaugeFactory;
-    });
-    testRelativeWeightCapForGauge('LiquidityGaugeV5');
+    testRelativeWeightCapForGauge(GaugeKind.LiquidityGauge);
   });
 
   describe('StakelessGauge', () => {
-    beforeEach('use stakeless gauge factory', () => {
-      factory = stakelessGaugeFactory;
-    });
-    testRelativeWeightCapForGauge('SingleRecipientGauge');
+    testRelativeWeightCapForGauge(GaugeKind.StakelessGauge);
   });
 
-  function testRelativeWeightCapForGauge(contractName: string) {
+  function testRelativeWeightCapForGauge(kind: GaugeKind) {
+    let factory: Contract;
+
+    sharedBeforeEach(async () => {
+      factory = kind == GaugeKind.LiquidityGauge ? liquidityGaugeFactory : stakelessGaugeFactory;
+    });
+
+    async function sendGaugeDeployTx(
```

Comment:
hmm how about `createGaugeTx` ?
I find it a bit hard to wrap my head around `send` and the return type of this helper without reading it hard.

Reply:
What do you find hard to understand about it?